File: services/Microservice.MapsPointsUser.Api/Model/Dto/RouteOptimizeResponseDto.cs
namespace Microservice.MapsPointsUser.Api.Model.Dto
{
    public class RouteOptimizeResponseDto
    {
        public string Polyline { get; set; } = string.Empty;
        public double TotalDistanceMeters { get; set; }
        public int TotalDurationSeconds { get; set; }
        public List<RouteStepDto> Steps { get; set; } = new();
    }

    public class RouteStepDto
    {
        public double DistanceMeters { get; set; }
        public int DurationSeconds { get; set; }
        public string Instruction { get; set; } = string.Empty;
        public double Lat { get; set; }
        public double Lng { get; set; }
    }
}

File: mobile-android/AndandoAndo_App/AndandoAndo_App/app/src/main/java/com/example/app_andando_ando/data/dto/mapUser/RouteOptimizeResponseDto.kt
package com.example.app_andando_ando.data.dto.mapUser

data class RouteStepDto(
    val distanceMeters: Double,
    val durationSeconds: Long,
    val instruction: String?,
    val lat: Double?,
    val lng: Double?
)

data class RouteOptimizeResponseDto(
    val polyline: String,
    val totalDistanceMeters: Double,
    val totalDurationSeconds: Long,
    val steps: List<RouteStepDto>
)
